### PR TITLE
MM-35677 Detox/E2E: Add e2e for channel link and scrolling

### PR DESCRIPTION
**Pull request by josephbaylon**
#### Summary
- Added tests for MM-T178, MM-T2970, MM-T3147 in
    - `detox/e2e/test/messaging/channel_link.e2e.js`
    - `detox/e2e/test/messaging/scrolling.e2e.js`

Note: Skipped MM-T178 as there's an open issue https://mattermost.atlassian.net/browse/MM-36532

#### Ticket Link
JIRA tickets:
- https://mattermost.atlassian.net/browse/MM-35677

Zephyr cases:
- [MM-T178](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T178)
- [MM-T2970](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2970)
- [MM-T3147](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3147)

#### Screenshots
Screen Shot 2021-06-17 at 11 16 51 AM
Screen Shot 2021-06-17 at 11 46 42 AM

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by josephbaylon** [comment]
@enahum @saturninoabril Please review. Thanks

**Review by enahum — APPROVED** [review]

File: detox/e2e/test/messaging/channel_link.e2e.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    });
+
+    afterAll(async () => {
+        await ChannelScreen.logout();
+    });
+
+    it('MM-T178 should be able to open channel by tapping on channel link from main channel', async () => {
+        // # Post a channel link
+        await goToChannel(townSquareChannel.display_name);
+        const channelLink = `http://localhost:8065/${testTeam.name}/channels/${testChannel.name}`;
```
Comment (saturninoabril) [inline detox/e2e/test/messaging/channel_link.e2e.js:47]: I'd recommend to use `siteUrl`?

**Review by saturninoabril — APPROVED** [review]
LGTM, except for one request. Feel free to merge aytime.

File: detox/e2e/test/messaging/channel_link.e2e.js (hunk 1 above)
Comment (josephbaylon) [reply detox/e2e/test/messaging/channel_link.e2e.js:47]: I tried adding `import testConfig from '@support/test_config';` but TSC is complaining

```
Checking for TSC
ERROR: Check TSC hints.
app/store/mmkv_adapter.ts(94,13): error TS2322: Type 'boolean | undefined' is not assignable to type 'Promise<boolean>'.
  Type 'undefined' is not assignable to type 'Promise<boolean>'.
```

I think for the boolean `siteUrl: process.env.SITE_URL || 'http://localhost:8065',`

I'm keeping the original for now as I'm not sure how to resolve it.

File: detox/e2e/test/messaging/channel_link.e2e.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+        // # Open channel screen
+        await ChannelScreen.open(user);
+    });
+
+    afterAll(async () => {
+        await ChannelScreen.logout();
+    });
+
+    it('MM-T178 should be able to open channel by tapping on channel link from main channel', async () => {
```
Comment (josephbaylon) [inline detox/e2e/test/messaging/channel_link.e2e.js:44]:
```suggestion
    it('MM-T2970 should be able to open channel by tapping on channel link from main channel', async () => {
```

File: detox/e2e/test/messaging/channel_link.e2e.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+// - Use element testID when selecting an element. Create one if none.
+// *******************************************************************
+
+import {ChannelScreen} from '@support/ui/screen';
+import {
+    Channel,
+    Post,
+    Setup,
+} from '@support/server_api';
+import {timeouts, wait} from '@support/utils';
```
Comment (josephbaylon) [inline detox/e2e/test/messaging/channel_link.e2e.js:16]:
```suggestion
import testConfig from '@support/test_config';
import {timeouts, wait} from '@support/utils';
```

File: detox/e2e/test/messaging/channel_link.e2e.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    });
+
+    afterAll(async () => {
+        await ChannelScreen.logout();
+    });
+
+    it('MM-T2970 should be able to open channel by tapping on channel link from main channel', async () => {
+        // # Post a channel link
+        await goToChannel(townSquareChannel.display_name);
+        const channelLink = `http://localhost:8065/${testTeam.name}/channels/${testChannel.name}`;
```
Comment (josephbaylon) [inline detox/e2e/test/messaging/channel_link.e2e.js:47]:
```suggestion
        const channelLink = `${testConfig.siteUrl}/${testTeam.name}/channels/${testChannel.name}`;
```

File: detox/e2e/test/messaging/channel_link.e2e.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        await wait(timeouts.ONE_SEC);
+
+        // * Verify redirected to test channel
+        await expect(channelNavBarTitle).toHaveText(testChannel.display_name);
+    });
+
+    xit('MM-T178 should be able to open channel by tapping on channel link from reply thread', async () => { // related issue https://mattermost.atlassian.net/browse/MM-36532
+        // # Post a channel link
+        await goToChannel(townSquareChannel.display_name);
+        const channelLink = `http://localhost:8065/${testTeam.name}/channels/${testChannel.name}`;
```
Comment (josephbaylon) [inline detox/e2e/test/messaging/channel_link.e2e.js:61]:
```suggestion
        const channelLink = `${testConfig.siteUrl}/${testTeam.name}/channels/${testChannel.name}`;
```

File: detox/e2e/test/messaging/channel_link.e2e.js (hunk 1 above)
Comment (josephbaylon) [reply detox/e2e/test/messaging/channel_link.e2e.js]: Done

**Pull request merged**